crates/node/src/storage/rocksdb.rs

```diff
@@ -218,21 +229,34 @@ pub fn open(
         REPLAY_PROTECTION_CF,
         replay_protection_cf_opts,
     ));
-
-    rocksdb::DB::open_cf_descriptors(&db_opts, path, cfs)
-        .map(RocksDB)
-        .map_err(|e| Error::DBError(e.into_string()))
+    Ok(if read_only {
```

Comment:
ah... nevermind. rip. I see how this is being used. we would need another field to block db writes of existing instances opened in read-write mode